### PR TITLE
[ports/libsigsegv] New port

### PR DESCRIPTION
**Pull request by SamuelMarks**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

Running it through your CI for Windows support, then will switch to draft?

Windows install instructions:
https://git.savannah.gnu.org/gitweb/?p=libsigsegv.git;a=blob;f=INSTALL.windows;h=b41d0123592a89f33595f395625752fe103333dc;hb=HEAD

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
Please get [failure logs here](https://dev.azure.com/vcpkg/public/_build/results?buildId=96656&view=logs&j=7922e5c4-0103-5f8f-ad17-45ce9bb98e80&t=66eef212-08d6-5e6c-a560-7d01149d0188).
```
Installing 1/1 libsigsegv:x64-windows...
Building libsigsegv:x64-windows...
-- Downloading https://ftp.gnu.org/gnu/libsigsegv/libsigsegv-2.14.tar.gz;https://www.mirrorservice.org/sites/ftp.gnu.org/gnu/libsigsegv/libsigsegv-2.14.tar.gz -> libsigsegv-2.14.tar.gz...
-- Extracting source D:/downloads/libsigsegv-2.14.tar.gz
-- Using source at D:/buildtrees/libsigsegv/src/bsigsegv-2-fd354ecde7.clean
-- Installing: D:/packages/libsigsegv_x64-windows/share/libsigsegv/copyright
-- Performing post-build validation
warning: The folder /include is empty or not present. This indicates the library was not correctly installed.
error: Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile: C:\a\1\s\ports\libsigsegv\portfile.cmake
error: building libsigsegv:x64-windows failed with: POST_BUILD_CHECKS_FAILED
```

**Comment by MonicaLiu0311** [comment]
Note: I will be converting your PR to draft status. When you're ready, please revert to "ready for review".

**Comment by MonicaLiu0311** [comment]
Is there any new progress?

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Comment by SamuelMarks** [comment]
@MonicaLiu0311 Reopen pushing some stuff up now

**Comment by MonicaLiu0311** [comment]
Please get [failure logs here](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/100076/artifacts?artifactName=failure%20logs%20for%20x64-windows&api-version=7.1&%24format=zip).
```
handler.c
stackvma.c
leave.c
dispatcher.c
version.c
D:\b\libsigsegv\src\bsigsegv-2-fd354ecde7.clean\src\handler-win32.c(217): error C2039: 'Esp': is not a member of '_CONTEXT'
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\winnt.h(4206): note: see declaration of '_CONTEXT'
D:\b\libsigsegv\src\bsigsegv-2-fd354ecde7.clean\src\handler-win32.c(219): error C2039: 'Eip': is not a member of '_CONTEXT'
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\winnt.h(4206): note: see declaration of '_CONTEXT'
make[2]: Leaving directory '/d/b/libsigsegv/x64-windows-dbg/src'
make[1]: Leaving directory '/d/b/libsigsegv/x64-windows-dbg
```

**Comment by MonicaLiu0311** [comment]
Ping @SamuelMarks

**Comment by MonicaLiu0311** [comment]
Is there any new progress?

**Comment by SamuelMarks** [comment]
@MonicaLiu0311 Why don't I just push it up without Windows support and add a new PR for Windows support (when I—or someone else—gets the time).

File: ports/libsigsegv/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "libsigsegv",
+  "version": "2.14.0",
+  "description": "GNU libsigsegv is a library for handling page faults in user mode.",
+  "homepage": "https://www.gnu.org/software/libsigsegv/",
+  "supports": "!uwp & !windows"
```
Comment (MonicaLiu0311) [inline ports/libsigsegv/vcpkg.json:6]: Hello, `"supports"` needs to be consistent with the platform supported by the upstream. If the upstream supports the windows platform but you are currently unable to solve the problems with the windows build, please add the failed [`triplets`](https://github.com/microsoft/vcpkg/tree/master/triplets) to [`ci.baseline.txt`](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt#L105-L108).

**Comment by MonicaLiu0311** [comment]
After installation, it is found that you have provided neither `.pc` files nor `config*.cmake` files. Please provide at least one `usage`.

File: ports/libsigsegv/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#            CFLAGS="-MD"
+#            CXX="$HOME/msvc/compile cl -nologo"
+#            CXXFLAGS="-MD"
+#            CPPFLAGS="-D_WIN32_WINNT=$win32_target -I/usr/local/msvc64/include"
+#            LDFLAGS="-L/usr/local/msvc64/lib"
+#            LD="link"
+#            NM="dumpbin -symbols"
+#            STRIP=":"
+#            AR="$HOME/msvc/ar-lib lib"
+#            RANLIB=":"
```
Comment (MonicaLiu0311) [inline ports/libsigsegv/portfile.cmake:32]: Are these comments necessary?

**Comment by MonicaLiu0311** [comment]
Note: I will be converting your PR to draft status. When you're ready, please revert to "ready for review".

**Comment by MonicaLiu0311** [comment]
Ping @SamuelMarks

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**